### PR TITLE
feat: 🎸 Add board schema

### PR DESCRIPTION
**Pull request by kamack38**

### CODE REVIEW

File: packages/shared/models/board.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import mongoose from 'mongoose'
+
+const boardSchema = new mongoose.Schema({
+  id: String,
+  title: String,
+  description: String,
+  createdAt: String,
+  admin: {
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:8]: admin is a user so we don't have to add separate field for that. What when there will be 2 admins? Remove

File: packages/shared/models/board.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  id: String,
+  title: String,
+  description: String,
+  createdAt: Date,
+  status: String,
+  users: [
+    {
+      id: String,
+      username: String,
+      avatarUrl: String
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:13]: what is this field for?
The images will be stored with type BSON. But it's to heavy to store it here
https://docs.mongodb.com/manual/reference/bson-types/

File: packages/shared/models/board.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  ],
+  columns: [
+    {
+      id: String,
+      title: String,
+      tasks: [
+        {
+          id: String,
+          title: String,
+          captionUrl: String,
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:24]: what is this field for?

File: packages/shared/models/board.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          comments: [{ id: String }],
+          attachments: [{ id: String }],
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:26]: so it means that we should create separate colleciotn for attachements?

File: packages/shared/models/board.ts (hunk 3 above)
Comment (kamack38) [reply packages/shared/models/board.ts:24]: This field is for board background image.

File: packages/shared/models/board.ts (hunk 4 above)
Comment (kamack38) [reply packages/shared/models/board.ts:26]: Yeah, I think so

File: packages/shared/models/board.ts (hunk 2 above)
Comment (mcwiekala) [reply packages/shared/models/board.ts]: change to avatarId: Number

File: packages/shared/models/board.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  ],
+  columns: [
+    {
+      id: String,
+      title: String,
+      tasks: [
+        {
+          id: String,
+          title: String,
+          captionUrl: String,
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:24]: change to: imageCoverId: Number

File: packages/shared/models/board.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import mongoose from 'mongoose'
+
+const boardSchema = new mongoose.Schema({
+  id: String,
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:4]: change this field to _id use type: ObjectId
https://www.tutorialspoint.com/mongodb/mongodb_objectid.htm
to create test data you can use generator:
https://observablehq.com/@hugodf/mongodb-objectid-generator
eg. _id: ObjectId('6225239aa76ad13dd37178b6'),

File: packages/shared/models/board.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  title: String,
+  description: String,
+  createdAt: Date,
+  status: String,
+  users: [
+    {
+      id: String,
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:11]: change this field to _id use type: ObjectId 
like comment above

File: packages/shared/models/board.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+      username: String,
+      avatarId: Number
+    }
+  ],
+  columns: [
+    {
+      id: String,
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:18]: change this field to _id use type: ObjectId 
like comment above

File: packages/shared/models/board.ts [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      title: String,
+      tasks: [
+        {
+          id: String,
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:22]: change this field to _id use type: ObjectId 
like comment above

File: packages/shared/models/board.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+          title: String,
+          imageCoverId: Number,
+          comments: [{ id: String }],
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:25]: change this field to _id use type: ObjectId 
like comment above

File: packages/shared/models/board.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+          attachments: [{ id: String }],
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:26]: change this field to _id use type: ObjectId 
like comment above

File: packages/shared/models/board.ts [hunk 12]
```diff
@@ -0,0 +1,39 @@
+import mongoose from 'mongoose'
```
[the remark this answers is not in the document]
Comment (kamack38) [reply packages/shared/models/board.ts:1]: Property 'ObjectId' does not exist on type 'typeof Schema'.ts(2339)

[the remark this answers is not in the document]
Comment (kamack38) [reply packages/shared/models/board.ts:1]:
```ts
Property 'ObjectId' does not exist on type 'typeof Schema'.ts(2339)
```

File: packages/shared/models/board.ts [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+      _id: ObjectId,
+      title: String,
+      tasks: [
+        {
+          _id: ObjectId,
+          title: String,
+          imageCoverId: Number,
+          comments: [{ _id: ObjectId }],
+          attachments: [{ _id: ObjectId }],
+          users: [{ id: String }]
```
[the remark this answers is not in the document]
Comment (mcwiekala) [reply packages/shared/models/board.ts:29]: sure

File: docker/init/02-boards.ts [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+          comments: [{ _id: new ObjectId('622524a5e8f9705974dbd045') }],
+          attachments: [{ _id: new ObjectId('491170ada2d211ecadb06e4d0c3fcdd4') }],
+          users: [{ _id: new ObjectId('622c727401dc22218e261681') }]
+        }
+      ]
+    }
+  ]
+}
+
+db.boards.insert(board1)
```
Comment (kamack38) [inline docker/init/02-boards.ts:67]:
```ts
Cannot find name 'db'.ts(2304)
```

File: docker/init/02-boards.ts [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+db.boards.insert(board2)
```
Comment (kamack38) [inline docker/init/02-boards.ts:68]:
```ts
Cannot find name 'db'.ts(2304)
```

**Review by kamack38 — COMMENTED** [review]
When doing the same thing as Ryszard I get a ts errors

File: packages/shared/models/board.ts [hunk 16]
```diff
@@ -0,0 +1,36 @@
+import mongoose from 'mongoose'
+
+const { ObjectId } = mongoose.Types
+
+const boardSchema = new mongoose.Schema({
```
Comment (mcwiekala) [inline packages/shared/models/board.ts:5]: What is our naming convention here? Board or Dashboard? Right now we have dashboard in project

Comment (kamack38) [reply packages/shared/models/board.ts:5]: Board is better and more descriptive

File: packages/backend/src/modules/dashboard/Board.ts [hunk 17]
```diff
@@ -0,0 +1,36 @@
+import mongoose from 'mongoose'
+
+const { ObjectId } = mongoose.Types
+
+const boardSchema = new mongoose.Schema({
+  _id: ObjectId,
+  title: String,
+  description: String,
+  createdAt: Date,
+  status: String,
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/Board.ts]: create enum for status: PUBLIC, PRIVATE
see
https://mongoosejs.com/docs/validation.html

File: packages/backend/src/modules/dashboard/Board.ts [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+  users: [
+    {
+      _id: ObjectId,
+      username: String,
+      avatarId: Number
+    }
+  ],
+  columns: [
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/Board.ts]: create schema for Column, use embeded document
see:
https://www.youtube.com/watch?v=b59Pdt5LJG8&ab_channel=WornOffKeys

**Pull request merged**